streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -578,4 +580,14 @@ private StateStoreMetadata findStore(final TopicPartition changelogPartition) {
 
         return found.isEmpty() ? null : found.get(0);
     }
+
+    // Pass in a sentinel value to checkpoint when the changelog offset is not yet initialized/known
+    private long checkpointableOffsetFromChangelogOffset(final Long offset) {
+        return offset != null ? offset : OFFSET_UNKNOWN;
```

Comment:
This is a little awkward, and it might be cleaner to just replace the use of `null` with this `OFFSET_UNKNOWN` sentinel throughout the ProcessorStateManager/StoreChangelogReader -- but, I wanted to keep the changes as short and simple as possible for now